pkg/sql/table.go

```diff
@@ -285,6 +285,20 @@ func (tc *TableCollection) getTableVersion(
 
 	origTimestamp := flags.txn.OrigTimestamp()
 	table, expiration, err := tc.leaseMgr.AcquireByName(ctx, origTimestamp, dbID, tn.Table())
+	if err == sqlbase.ErrDescriptorNotFound {
+		// The table doesn't exist at the requested timestamp.
+		// Read the table descriptor directly to check if the descriptor
+		// exists at a future timestamp within the clock uncertainty
+		// interval.
+		uncachedFlags := flags
+		uncachedFlags.avoidCached = true
+		phyAccessor := UncachedPhysicalAccessor{}
+		_, _, errRetry := phyAccessor.GetObjectDesc(tn, uncachedFlags)
+		if errRetry == nil && flags.txn.Proto().RefreshedTimestamp != (hlc.Timestamp{}) {
```

Comment:
This seems pretty iffy. Let's discuss in person tomorrow – there must be a better way to do this than breaking the abstraction pretty thoroughly. Code at this level should not be peering into the transaction proto to make decisions.

Reply:
As I was telling Vivek earlier, I think the real problem here is that `leaseMgr.AcquireByName()` basically swallows the uncertainty of whatever read it does and lets higher layers fend for themselves. I'll attempt to do something about that as soon as I can, unless Vivek does something else first.